### ISSUE TITLE
QR code fails to load on 2FA setup page

### ISSUE THREAD

**Issue by KingTeppic** [MbinOrg/mbin#159]
As per this post:
https://fedia.io/m/fedia/t/369187
Two Factor Authentication setup process is currently broken.

The QR code doesn't show (browser puts an icon to indicate the image source is broken)

I have verified this issue is present on both Melroy's instance and fedia.io

To recreate:
Navigate to password settings, e.g.
https://fedia.io/settings/password
Click "Setup two-factor authentication"

There should be a QR code which fails to show. 
Worryingly (for users) on the same page it also advised the displayed recovery codes will not been shown again...

This is all as observed on mobile, I've not tried on a desktop.

**Comment by e-five256** [comment]
I can confirm it gives a 404 nginx on fedia and melroy's instance, but for my dev env and on `kbin.run` it appears to work without issue. @melroy89 if you get some time could you maybe check it out on your env and maybe see if it's the same issue fedia is having?

**Comment by melroy89** [comment]
Sure I will check today. Maybe some missing configuration. Let me debug this soon. If the root cause is found I also update either the code or documentation.

**Comment by e-five256** [comment]
> Worryingly (for users) on the same page it also advised the displayed recovery codes will not been shown again...

Regarding this, it appears that it's true you can't view them again, but you can generate new ones instead, so perhaps some copy change is necessary to make that more clear

**Comment by melroy89** [comment]
Pretty sure it's a Nginx misconfiguration by now.

**Comment by melroy89** [comment]
My proposal is to remove the `png` extension from the code and frontend, because it's a dynamic generated image. Using this `.png` at the of the path causes all kind of image optimization paths in Nginx or even CDNs.

**Comment by melroy89** [comment]
PR created this will resolve the problem once and for all: https://github.com/MbinOrg/mbin/pull/164

**Comment by KingTeppic** [comment]
Much lower priority, but it might be good to change the warning to make it clear you can safely come back to the page at a later date.

Currently it says:

> You can use these codes when you don't have your two-factor authentication device or app. You will not be shown them again and will be able to use each of them only once.
> 
> It is recommended that you keep a copy of them in a safe place.

Which perhaps feels like "now you've started, this is the only chance you get!"
...perhaps change this to:

> You can use these codes when you don't have your two-factor authentication device or app. 
> These backup codes are paired with the QR code above - they include randomisation which will change each time you view this page. 
> 
> When enabling two-factor authentication it is recommended that you keep a copy of the backup codes in a safe place. You will not be shown these codes again and you will only be able to use each recovery code once.

...but that's a long way down the list of 'nice to have'.
